devlib/utils/ssh.py

```diff
@@ -695,6 +695,9 @@ def _gem5_shell(self, command, as_root=False, timeout=None, check_exit_code=True
 
         gem5_logger.debug("gem5_shell command: {}".format(command))
 
+        if as_root:
+            command = 'echo "{}" | su'.format(command)
```

Comment:
Should be using `escape_double_quotes()` on the `command` to make sure the echo quote isn't closed early if there happen to be a double quote character in the command itself.